### ISSUE TITLE
Support Samba4 AD ldap scheme without "email" attribute

### ISSUE THREAD

**Issue by tanertas** [Requarks/wiki#695]
### Actual behavior
Wiki.js doesn't allow login .
### Expected behavior
Wiki.js should map "userprincipalname" attribute to "email" thus allow AD setups without "email" attribute in their scheme.
### Steps to reproduce the behavior
Configure wiki for ldap auth using Samba4 AD with no "email" attribute then try to login.

**Comment by NGPixel** [comment]
Does `userprincipalname` contain an email though? It can be located in any field, but an email is required.

**Comment by tanertas** [comment]
Yes, userprincipalname contains an email for related domain which is basically in form of samaccountname@fqdn. I think we should at least have a config option for which ldap field is going to be assigned for "email".
